src/structures/MessageEmbed.js

```diff
@@ -304,7 +304,7 @@ class MessageEmbed {
    * @returns {MessageEmbed}
    */
   setFields(...fields) {
-    this.fields = this.constructor.normalizeFields(fields);
+    this.spliceFields(0, this.fields.length, this.constructor.normalizeFields(fields));
```

Comment:
`MessageEmbed#spliceFields` is already normalizing the fields:
```suggestion
    this.spliceFields(0, this.fields.length, fields);
```